app/helpers/template_helpers.py

```diff
@@ -1,304 +1,379 @@
-    default_context = {
-        "logo": "ons-logo-pos-" + language,
-        "logoAlt": lazy_gettext("Office for National Statistics logo"),
-    }
-    context = {
-        "default": default_context,
-        "social": default_context,
-        "northernireland": default_context,
-        "census": {
-            **default_context,
-            "titleLogo": f"census-logo-{language}",
-            "titleLogoAlt": lazy_gettext("Census 2021"),
-        },
-        "census-nisra": {
-            "logo": "nisra-logo-en",
-            "mobileLogo": "nisra-logo-en-mobile",
-            "logoAlt": lazy_gettext(
-                "Northern Ireland Statistics and Research Agency logo"
-            ),
-            "titleLogo": "census-logo-en",
-            "titleLogoAlt": lazy_gettext("Census 2021"),
-            "customHeaderLogo": "nisra",
-        },
-    }
-    return context.get(theme, {})
-
-
-def get_footer_context(
-    language_code: str, static_content_urls: Mapping, sign_out_url: str, theme: str
-) -> Optional[Dict[str, Any]]:
-
-    items_list = [
-        {
-            "text": lazy_gettext("Help"),
-            "url": static_content_urls["help"],
-            "target": "_blank",
-        },
-        {
-            "text": lazy_gettext("Contact us"),
-            "url": static_content_urls["contact_us"],
-            "target": "_blank",
-        },
-    ]
-    if theme != "census-nisra":
-        items_list.extend(
-            [
-                {
-                    "text": lazy_gettext("Languages"),
-                    "url": static_content_urls["languages"],
-                    "target": "_blank",
-                },
-                {
-                    "text": lazy_gettext("BSL and audio videos"),
-                    "url": static_content_urls["bsl_and_audio_videos"],
-                    "target": "_blank",
-                },
-            ]
+
+@dataclass
+class ContextOptions:
+    """Valid options for defining context."""
+
+    page_header_logo: Optional[str] = "ons-logo-pos-en"
+    page_header_logo_alt: Optional[LazyString] = lazy_gettext(
+        "Office for National Statistics logo"
+    )
+    copyright_declaration: Optional[LazyString] = lazy_gettext(
+        "Crown copyright and database rights 2020 OS 100019153."
+    )
+    copyright_text: Optional[LazyString] = lazy_gettext(
+        "Use of address data is subject to the terms and conditions."
+    )
+    base_url: str = CENSUS_EN_BASE_URL
+    account_service_url: Optional[str] = None
+    title_logo: Optional[str] = None
+    title_logo_alt: Optional[str] = None
+    header_logo: Optional[str] = None
+    mobile_logo: Optional[str] = None
+    powered_by_logo: Optional[str] = None
+    powered_by_logo_alt: Optional[str] = None
+    crest: bool = True
+    footer_links: Optional[Iterable[MutableMapping]] = None
+    footer_legal_links: Optional[Iterable[Mapping]] = None
+    survey_title: Optional[LazyString] = lazy_gettext("Census 2021")
```

Comment:
This shouldn't have a default of `Census 2021`, it should come from the derived options.

Reply:
Done